xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/WCAGContext.java

```diff
@@ -452,16 +537,72 @@ public String buildWarningsReport()
+            overview.append(System.lineSeparator());
+            overview.append(String.format("  [%d](%.1f%%) were incomplete",
+                this.wcagIncompleteCount, 100 * (float) this.wcagIncompleteCount / totalChecks));
+            overview.append(System.lineSeparator());
+            overview.append(String.format("  [%d](%.1f%%) were passed",
+                    this.wcagPassCount, 100 * (float) this.wcagPassCount / totalChecks));
+
+
+            buildResultStatusOverview(overview, this.wcagViolationCount,
+                    "Here a list of the rule IDs, sorted by the number of failed checks:",
+                    this.getViolationCountPerRule(), "FAIL");
+            buildResultStatusOverview(overview, this.wcagIncompleteCount,
+                    "Here a list of the rule IDs, sorted by the number of incomplete checks:",
+                    this.getIncompleteCountPerRule(), "INCOMPLETE");
+            buildResultStatusOverview(overview, this.wcagPassCount,
+                    "Here a list of the rule IDs, sorted by the number of passed checks:",
+                    this.getPassCountPerRule(), "PASS");
+        }
+        overview.append(System.lineSeparator());
+        return overview.toString();
+    }
+
+    private void buildResultStatusOverview(StringBuilder overview, Integer statusCount, String descriptionString,
+        Map<String, Integer> statusCounts, String statusID)
+    {
+        if (statusCount > 0) {
+            overview.append(System.lineSeparator());
+            overview.append(System.lineSeparator());
+            overview.append(descriptionString);
+            Ordering.natural().onResultOf(Functions.forMap(statusCounts));
+            statusCounts.entrySet().stream()
+                .sorted(Map.Entry.<String, Integer>comparingByValue().reversed())
+                .forEach(entry -> {
+                    overview.append(System.lineSeparator());
+                    overview.append(String.format("  %s %s [%d](%.1f%%)",
+                        statusID, entry.getKey(), entry.getValue(),
+                        100 * (float) entry.getValue() / allCountPerRule.get(entry.getKey())));
+                });
+        }
+    }
+
     /**
      * Writes a WCAG report to a file for proper accessibility warnings and failures examination.
      *
      * @param outputFile the file to write the test results to
      * @param output the test results to write
      */
-    public static void writeWCAGReportToFile(final File outputFile, final String output) throws IOException
+    public static void writeWCAGReportToFile(File outputFile, String output) throws IOException
     {
-        Writer writer = new OutputStreamWriter(
-                    new FileOutputStream(outputFile), StandardCharsets.UTF_8);
+        FileWriter writer = new FileWriter(outputFile, StandardCharsets.UTF_8);
         writer.write(output);
+        writer.close();
```

Comment:
```suggestion
        try (FileWriter writer = new FileWriter(outputFile, StandardCharsets.UTF_8)) {
            writer.write(output);
        }
```

Using a [try-with-resource statement](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) is cleaner and avoids issues if there should be any exception during writing.